app/models/mixins/event_mixin.rb

```diff
@@ -36,10 +36,39 @@ def events_table_name(assoc)
     events_assoc_class(assoc).table_name
   end
 
+  def event_stream_filters
+    {
+      "EmsEvent".freeze => ems_event_filter,
+      "MiqEvent".freeze => miq_event_filter
+    }
+  end
+
+  private
+
+  def ems_event_filter
+    {self.class.ems_event_filter_column => id}
+  end
+
+  def miq_event_filter
+    filter = {self.class.miq_event_filter_column => id}
+    filter["target_type"] = self.class.base_class.name if filter.key?("target_id")
+    filter
+  end
+
   private
 
   def find_one_event(assoc, order)
     ewc = event_where_clause(assoc)
     events_assoc_class(assoc).where(ewc).order(order).first unless ewc.blank?
   end
+
+  module ClassMethods
+    def ems_event_filter_column
+      @ems_event_filter_column ||= reflect_on_association(:ems_events).try(:foreign_key) || name.foreign_key
+    end
+
+    def miq_event_filter_column
+      @miq_event_filter_column ||= reflect_on_association(:miq_events).try(:foreign_key) || "target_id".freeze
```

Comment:
See the table in the bottom of the description for more details as to how I came up with this.  It's the closest to what we do with `event_where_clause`.

Reply:
Note, I need to fix the outliers, such as one that also add another column for their specific class (container_name):

```ruby
app/models/container.rb:  def event_where_clause(assoc = :ems_events)
app/models/container.rb-    case assoc.to_sym
app/models/container.rb-    when :ems_events, :event_streams
app/models/container.rb-      # TODO: improve relationship using the id
app/models/container.rb-      ["container_namespace = ? AND #{events_table_name(assoc)}.ems_id = ? AND container_name = ?",
app/models/container.rb-       container_project.try(:name), ext_management_system.try(:id), name]
app/models/container.rb-    when :policy_events
app/models/container.rb-      # TODO: implement policy events and its relationship
app/models/container.rb-      ["#{events_table_name(assoc)}.ems_id = ?", ext_management_system.try(:id)]
app/models/container.rb-    end
app/models/container.rb-  end
```

Reply:
I think I'll do that as a followup as we don't even have container data to test with.